src/ol_infrastructure/infrastructure/mongodb/__main__.py

```diff
@@ -0,0 +1,209 @@
+import json
+from itertools import chain
+
+from pulumi import Config, ResourceOptions
+from pulumi.stack_reference import StackReference
+from pulumi_aws import ec2, iam
+
+from ol_infrastructure.lib.aws.ec2_helper import (
+    InstanceTypes,
+    build_userdata,
+    debian_10_ami,
+    default_egress_args,
+)
+from ol_infrastructure.lib.aws.iam_helper import lint_iam_policy
+from ol_infrastructure.lib.ol_types import AWSBase
+from ol_infrastructure.lib.pulumi_helper import parse_stack
+from ol_infrastructure.providers.salt.minion import (
+    OLSaltStackMinion,
+    OLSaltStackMinionInputs,
+)
+
+stack_info = parse_stack()
+env_config = Config("environment")
+mongodb_config = Config("mongodb")
+salt_config = Config("saltstack")
+environment_name = f"{stack_info.env_prefix}-{stack_info.env_suffix}"
+business_unit = env_config.get("business_unit") or "operations"
+aws_config = AWSBase(tags={"OU": business_unit, "Environment": environment_name})
+network_stack = StackReference(f"infrastructure.aws.network.{stack_info.name}")
+operations_vpc = network_stack.require_output("operations_vpc")
+consul_stack = StackReference(
+    "infrastructure.consul."  # noqa: WPS237, WPS221
+    f"{stack_info.namespace.rsplit('.', 1)[1]}."  # noqa: WPS237, WPS221
+    f"{stack_info.name}"
+)
+destination_vpc = network_stack.require_output(env_config.require("vpc_reference"))
+
+mongodb_instance_policy = {
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Action": [
+                "ec2:DescribeInstances",
+                "ec2:DescribeAvailabilityZones",
+                "ec2:DescribeRegions",
+                "ec2:DescribeSecurityGroups",
+                "ec2:DescribeTags",
+            ],
+            "Effect": "Allow",
+            "Resource": ["*"],
+        },
+    ],
+}
+
+mongodb_iam_policy = iam.Policy(
+    f"mongodb-policy-{environment_name}",
+    name=f"mongodb-policy-{environment_name}",
+    path=f"/ol-applications/mongodb-{environment_name}/",
+    policy=lint_iam_policy(mongodb_instance_policy, stringify=True),
+    description="Policy to access resources needed by mongodb instances",
+)
+
+mongodb_instance_role = iam.Role(
+    "mongodb-instance-role",
+    assume_role_policy=json.dumps(
+        {
+            "Version": "2012-10-17",
+            "Statement": {
+                "Effect": "Allow",
+                "Action": "sts:AssumeRole",
+                "Principal": {"Service": "ec2.amazonaws.com"},
+            },
+        }
+    ),
+    name=f"mongodb-instance-role-{environment_name}",
+    path=f"/ol-operations/mongodb-{environment_name}/",
```

Comment:
And the path here is `ol-operations`. We should probably define a specific naming schema and enforce it via code for defining these paths.